envoy-control-core/src/main/resources/lua/ingress_client_name_header.lua

```diff
@@ -0,0 +1,25 @@
+function envoy_on_request(handle)
+    local streamInfo = handle:streamInfo()
+    local trusted_header = handle:metadata():get("trusted_client_identity_header") or ""
+    if trusted_header == "" then
+        return
+    end
+
+    if handle:headers():get(trusted_header) ~= nil then
```

Comment:
Will remove of a header that does not exist cause troubles? Maybe we could omit the check?

Reply:
I asked the same question when I saw this code. I checked envoy sources, and it will definitely work without a check. But I'm not sure how it will be performance-wise, so I kept the check.